contracts/lazyclaim/ERC721LazyClaim.sol

```diff
@@ -0,0 +1,159 @@
+    uint walletMax,
+    uint startDate,
+    uint endDate
+  ) external creatorAdminRequired(creatorContractAddress) {
+    // Overwrite the existing claim
+    claims[creatorContractAddress][index] = Claim({
+      merkleRoot: merkleRoot,
+      uri: uri,
+      totalMax: totalMax,
+      walletMax: walletMax,
+      startDate: startDate,
+      endDate: endDate
+    });
+  }
+  function setMerkleRoot(address creatorContractAddress, uint index, bytes32 merkleRoot) external {
+    Claim memory claim = claims[creatorContractAddress][index];
+    claim.merkleRoot = merkleRoot;
+    claims[creatorContractAddress][index] = claim;
+  }
+  function setURI(address creatorContractAddress, uint index, string calldata uri) external {
+    Claim memory claim = claims[creatorContractAddress][index];
+    claim.uri = uri;
+    claims[creatorContractAddress][index] = claim;
+  }
+  function setTotalMax(address creatorContractAddress, uint index, uint totalMax) external {
+    Claim memory claim = claims[creatorContractAddress][index];
+    claim.totalMax = totalMax;
+    claims[creatorContractAddress][index] = claim;
+  }
+  function setWalletMax(address creatorContractAddress, uint index, uint walletMax) external {
+    Claim memory claim = claims[creatorContractAddress][index];
+    claim.walletMax = walletMax;
+    claims[creatorContractAddress][index] = claim;
+  }
+  function setStartDate(address creatorContractAddress, uint index, uint startDate) external {
+    Claim memory claim = claims[creatorContractAddress][index];
+    claim.startDate = startDate;
+    claims[creatorContractAddress][index] = claim;
+  }
+  function setEndDate(address creatorContractAddress, uint index, uint endDate) external {
+    Claim memory claim = claims[creatorContractAddress][index];
+    claim.endDate = endDate;
+    claims[creatorContractAddress][index] = claim;
+  }
+
+  // Public getters
+  function getClaimCount(address creatorContractAddress) external view returns(uint) {
+    return claimCounts[creatorContractAddress];
+  }
+  function getClaim(address creatorContractAddress, uint index) external view returns(Claim memory) {
+    return claims[creatorContractAddress][index];
+  }
+
+  // Public mint
+  function mint(address creatorContractAddress, uint index, bytes32[] calldata merkleProof) external {
+      // Verify merkle proof
+      Claim memory claim = claims[creatorContractAddress][index];
+      if (claim.merkleRoot != "") {
+        bytes32 leaf = keccak256(abi.encodePacked(msg.sender));
+        require(MerkleProof.verify(merkleProof, claim.merkleRoot, leaf), "Could not verify merkle proof");
+      }
+
+      // Check walletMax against minter's wallet
+      uint walletMinted = mintsPerWallet[creatorContractAddress][index][msg.sender];
+      if (claim.walletMax != 0) require(walletMinted < claim.walletMax, "Maximum tokens already minted for this wallet");
+      mintsPerWallet[creatorContractAddress][index][msg.sender] = walletMinted + 1;
```

Comment:
Defer increments until the end (right before mint) just in case there's a further revert).  Do an uncekced ++ rather than +1 to save gas

Basically don't write until all checks are complete

Reply:
Done